src/Mod/Arch/ArchWall.py

```diff
@@ -600,8 +604,18 @@ def execute(self,obj):
                 for b in bplates:
                     b.Placement = extdata[2].multiply(b.Placement)
                     b = b.extrude(extv)
-                    shps.append(b)
-                base = Part.makeCompound(shps)
+
+                    # See getExtrusionData() - not fusing baseplates there - fuse solids here
+                    # Remarks - If solids are fused, but exportIFC.py use underlying baseplates w/o fuse, the result in ifc look slightly different from in FC.
+                    #shps.append(b)
+                    if shps:
+                        shps = shps.fuse(b) #shps.fuse(b)
+                    else:
+                        shps=b
+                    # TODO - To let user to select whether to fuse (slower) or to do a compound (faster) only ?
+                #base = Part.makeCompound(shps)
+                base = shps
+
```

Comment:
Do we ALWAYS want to fuse wall solids? I think in many cases not (ex. blocks...)

Reply:
That's one thing I am not sure and make a remark in line 615.  The original code try to fuse the baseplates and remove splitter but somehow this no longer works at some point especially for sketch.  So fusing the resultant solids seem a natural step.

I have another look and noted the calculation of blocks come after the original code making a compound of the shapes.  So not sure does it matter?

Probably like you says, may be just usecases with sketch needs fusing baseplates / solids, so may better introducing an option for user to choose similar to the Sketch.Refine options, default being no fusing is required ?

Reply:
That, or we detect special cases, for ex: Our wall is based on a single sketch, but it has many solids. In that case, I suppose it's safe to fuse...

Reply:
Thanks, will try to do it.

And an option to not fusing the solid making ArchWall output in FC same as IfcExport may be useful - unless finding way to fuse baseplate to make both output consistent.

Also checking addition, subtraction, layers in the meantime ... interesting that addition do some cut, layer get in the way doing basewire etc.  These are for later anyway.

Reply:
Add Test : if Sketch, fuse solids; otherwise, do makeCompound.

BTW, do some rough - seem no much difference doing fuse and makeCompound?

Reply:
Okay, seems good, thanks!
You won't see much difference in most cases, but if you have touching or intersecting solids, that's where it will make a difference.